jax/_src/xla_bridge.py

```diff
@@ -254,6 +255,64 @@ def make_cpu_client() -> xla_client.Client:
     "cpu", make_cpu_client, priority=0, fail_quietly=False
 )
 
+def _check_ptxas_compat() -> None:
+  import re
+  import subprocess
+  from glob import glob
+
+  cuda_path = None
+
+  # The pip-installed nvidia packages may have been installed in a separate
+  # location from JAX so we check for it specifically.
+  try:
+    import nvidia # type: ignore
+    nvidia_path = Path(nvidia.__file__).parent
+    cuda_path = nvidia_path.joinpath(Path("cuda_nvcc/bin"))
+  except ModuleNotFoundError:
+    pass
+
+  searchable_paths = [cuda_path] if cuda_path else []
+  from_PATH = os.environ["PATH"].split(os.pathsep)
+  searchable_paths += [Path(path) for path in from_PATH]
+  found_ptxas = None
+  for path in searchable_paths:
+      ptxas_relpath = glob("ptxas", root_dir=path)
+      if ptxas_relpath:
+        found_ptxas = path / Path(ptxas_relpath[0])
+        break
+
+  cuda_major = str(cuda_versions.cuda_runtime_build_version())[:2]
+  nvcc_pack = f"nvidia-cuda-nvcc-cu{cuda_major}"
+  if found_ptxas is None:
+    raise RuntimeError(
+      f"ptxas binary not found -- please either install `{nvcc_pack}` via "
+      f"`pip install {nvcc_pack}`, or add the directory containing your ptxas "
+      "binary to your PATH environment variable."
+    )
+
+  args = (found_ptxas, "--version")
+  version_info = subprocess.run(args, capture_output=True).stdout.decode()
+  pattern = re.compile("release (.*?),")
+  match = re.search(pattern, version_info)
+  if match is None:
+    raise RuntimeError(
+      f"The ptxas found at {found_ptxas} could not be invoked -- if using a "
+      "pip-provided ptxas, please try re-installing. If using a PATH-specified "
+      "ptxas, please correct your PATH variable to include a directory "
+      "containing a viable ptxas binary."
+    )
+  major, minor = match[1].split('.')
+  ptxas_version = int(major) * 1000 + int(minor) * 10
+  min_cuda_version = cuda_versions.cuda_runtime_build_version()
```

Comment:
Actually, we've subsequently relaxed this constraint. We build against 12.3, but allow deployment against 12.1 or newer.

I'd just hardwire in that version here.